src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTaggerProvider.cs

```diff
@@ -27,55 +28,52 @@ namespace Microsoft.CodeAnalysis.Editor.InlineHints
+        Lazy<IStreamingFindUsagesPresenter> streamingFindUsagesPresenter,
+        EditorOptionsService editorOptionsService,
+        TaggerHost taggerHost,
+        [Import(AllowDefault = true)] IInlineHintKeyProcessor inlineHintKeyProcessor) : IViewTaggerProvider
     {
-        private readonly IViewTagAggregatorFactoryService _viewTagAggregatorFactoryService;
-        public readonly IClassificationFormatMapService ClassificationFormatMapService;
-        public readonly IClassificationTypeRegistryService ClassificationTypeRegistryService;
-        public readonly IThreadingContext ThreadingContext;
-        public readonly IUIThreadOperationExecutor OperationExecutor;
-        public readonly IAsynchronousOperationListener AsynchronousOperationListener;
-        public readonly IToolTipService ToolTipService;
-        public readonly ClassificationTypeMap TypeMap;
-        public readonly Lazy<IStreamingFindUsagesPresenter> StreamingFindUsagesPresenter;
-        public readonly EditorOptionsService EditorOptionsService;
+        public readonly IGlobalOptionService GlobalOptionService = globalOptionService;
+        public readonly IClassificationFormatMapService ClassificationFormatMapService = classificationFormatMapService;
+        public readonly IClassificationTypeRegistryService ClassificationTypeRegistryService = classificationTypeRegistryService;
+        public readonly IThreadingContext ThreadingContext = threadingContext;
+        public readonly IUIThreadOperationExecutor OperationExecutor = operationExecutor;
+        public readonly IAsynchronousOperationListener AsynchronousOperationListener = listenerProvider.GetListener(FeatureAttribute.InlineHints);
+        public readonly IToolTipService ToolTipService = toolTipService;
+        public readonly ClassificationTypeMap TypeMap = typeMap;
+        public readonly Lazy<IStreamingFindUsagesPresenter> StreamingFindUsagesPresenter = streamingFindUsagesPresenter;
+        public readonly EditorOptionsService EditorOptionsService = editorOptionsService;
 
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public InlineHintsTaggerProvider(
-            IViewTagAggregatorFactoryService viewTagAggregatorFactoryService,
-            IClassificationFormatMapService classificationFormatMapService,
-            IClassificationTypeRegistryService classificationTypeRegistryService,
-            IThreadingContext threadingContext,
-            IUIThreadOperationExecutor operationExecutor,
-            IAsynchronousOperationListenerProvider listenerProvider,
-            IToolTipService toolTipService,
-            ClassificationTypeMap typeMap,
-            Lazy<IStreamingFindUsagesPresenter> streamingFindUsagesPresenter,
-            EditorOptionsService editorOptionsService)
+        private readonly InlineHintsDataTaggerProvider _dataTaggerProvider = new(taggerHost, inlineHintKeyProcessor);
```

Comment:
@akhera99  THis was the suggestion i had before.  Where we do not use a tag aggregator, we just have the inlint hints tagger call into the underlying data tagger provider.  I thought you were going to make that change.